scipy/stats/tests/test_stats.py

```diff
@@ -7927,6 +7927,37 @@ def test_basic(self):
         assert_equal(F, 2.0)
         assert_allclose(p, 1 - np.sqrt(0.5), rtol=1e-14)
 
+    def test_unequal_var(self):
+        # toy samples with unequal variances and different observations
+        samples = [[-50.42, 40.31, -18.09, 35.58, -6.8, 0.22],
+                   [23.44, 4.5, 15.1, 9.66],
+                   [11.94, 11.1 , 9.87, 9.09, 3.33]]
+
+        with assert_raises(TypeError):
+            stats.f_oneway(*samples, equal_var="False")
```

Comment:
Please make this a separate test - this part is input validation of the `equal_var` parameter; the rest of the test is accuracy of the Welch test. We also try to check for the particular error message these days (`match` parameter).

```python3
def test_equal_var_input_validation(self):
    message = "Expected a boolean value for 'equal_var'"
    with pytest.raises(TypeError, match=message):
        stats.f_oneway(*samples, equal_var="False")
```